R/kartverket-utils.R

```diff
@@ -200,13 +201,24 @@ parse_kartverket <- function(resp, parsed, x, radius, crs, ...) {
         tmp_parsed <- vector('list', iter+1)
         tmp_resp[[1]] <- resp
         tmp_parsed[[1]] <- parsed
-        for (i in seq_len(iter)) {
-          tmp_resp[[i+1]] <-
-            request_kartverket(
-              'punktsok', lat = x[1], lon = x[2], radius = radius,
-              crs = crs, page = i, ...)
-          tmp_parsed[[i+1]] <-
-            parse_response(tmp_resp[[i+1]], simplifyVector = TRUE)
+        if (!is.null(search)) {
+          # sok endpoint
+          for (i in seq_len(iter)) {
+            tmp_resp[[i+1]] <-
+              request_kartverket('sok', search, crs = crs, page = i, ...)
+            tmp_parsed[[i+1]] <-
+              parse_response(tmp_resp[[i+1]], simplifyVector = TRUE)
+          }
```

Comment:
You are growing a list with a for-loop. It would be more performant to use `lapply` and then merge the elements from the first page.

Reply:
Actually, I'm not sure that's correct. R's internal memory management was much improved in R 3.x. Particularly the problem with deep copies inside loops has been largely removed. So, at least ceteris paribus, there are no longer any significant perfomance differences between loops and apply-family functions. Also, please note that the length of the lists we are assigning to have been pre-defined (with `vector` in the code above). This should also help in terms of memory management. Finally I would guess that the most time-consuming part here is the _i_ number of additional API calls, not R code it self. 

I think a better argument for `lapply` is the fact that it fits more closely with a functional programming style. `parse_kartverket()` is at bit complex and it is certainly not obivious from the name that it also make additional API calls. So open to a different implementation or improvment at a later stage, but I think we can live with the current one for now.

Reply:
I agree. We close this for now, and revisit this issue at a later stage.